src/ui/common/formik-field/SwitchInput.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Switch, Col, ControlLabel } from 'patternfly-react';
+import { getTouchErrorByField } from 'helpers/formikUtils';
+
+const switchField = (
+  field, form, switchValue, trueValue,
+  falseValue, onToggleValue, dataTestId,
+) => {
+  const handleChange = (el, val) => {
+    const returnVal = val ? trueValue : falseValue;
+    form.setFieldValue(field.name, returnVal);
+    if (onToggleValue) {
+      onToggleValue(returnVal);
+    }
+  };
+
+  return (
+    <div data-testid={dataTestId} aria-labelledby={`switch-${field.name}`}>
+      <Switch
+        {...field}
+        value={switchValue}
+        onChange={handleChange}
+      />
+    </div>
+  );
+};
+
+const SwitchInput = ({
+  field, form, append, label, labelSize, inputSize, alignClass,
+  help, trueValue, falseValue, disabled, onToggleValue,
+}) => {
+  const switchValue = field.value === 'true' || field.value === true || field.value === trueValue;
+  const dataTestId = `${field.name}-switchField`;
+  const { touched, error } = getTouchErrorByField(field.name, form);
+
+  if (label) {
+    return (
+      <div className={`SwitchInput ${(touched && error) ? 'form-group has-error' : 'form-group'}`}>
+        <Col xs={labelSize} className={alignClass}>
+          <ControlLabel id={`switch-${field.name}`} htmlFor={field.name}>
+            {label} {help}
+          </ControlLabel>
+        </Col>
+        <Col xs={inputSize || 12 - labelSize}>
+          {switchField(
```

Comment:
is there any specific reason for `switchField` not be a React component?

Reply:
Just copied the existing `SwitchRenderer` component to the `formik-field` folder and amended it a bit to make it work with formik. So didn't touch those parts of the code. But indeed, not sure why it was written that way. Making the changes wouldn't take long probably so will do that.